rest-server/src/models/job.js

```diff
@@ -365,7 +365,7 @@ class Job {
   generateFrameworkDescription(data) {
     const gpuType = data.gpuType || null;
     const fancyRetryPolicy = (data.retryCount >= -1);
-    const killOnCompleted = (data.killAllOnCompletedTaskNumber > 0);
+    const minSucceededTaskCount = (data.killAllOnCompletedTaskNumber > 0) ? 1 : null;
```

Comment:
`minSucceededTaskCount` = `data.killAllOnCompletedTaskNumber`?

Reply:
killAllOnAnyCompleted is equivalent to
"applicationCompletionPolicy": {
          "minFailedTaskCount": 1,
          "minSucceededTaskCount": 1
        },

---
In reply to: [196669188](https://github.com/Microsoft/pai/pull/742#discussion_r196669188) [](ancestors = 196669188)

Reply:
minSucceededTaskCount should be exactly equals to 1 to achieve the same behavior as killAllOnAnyCompleted 

---
In reply to: [196669668](https://github.com/Microsoft/pai/pull/742#discussion_r196669668) [](ancestors = 196669668,196669188)

Reply:
I see. Should `minSucceededTaskCount` be `0` or `null` to disable `killAllOnAnyCompleted`?

Reply:
For how to refine RestServer later, please check user manual:

<tr>
      <td rowspan="2"><b>TensorFlow</td>
      <td>ParameterServer</td>
      <td><i>MinFailedTaskCount = 1<br>MinSucceededTaskCount = null</i></td>
      <td rowspan="2">Succeed a certain TaskRole is enough, and do not want to wait all Tasks to succeed:<br>Fail the Application immediately if any Task failed.<br>Succeed the Application immediately if Worker's all Tasks succeeded.</td>
    </tr>
    <tr>
      <td>Worker</td>
      <td><i>MinFailedTaskCount = 1</i><br>MinSucceededTaskCount = {Worker.TaskNumber}</td>
    </tr>


---
In reply to: [196669922](https://github.com/Microsoft/pai/pull/742#discussion_r196669922) [](ancestors = 196669922,196669668,196669188)

Reply:
Should be null to disable killAllOnAnyCompleted

---
In reply to: [196670423](https://github.com/Microsoft/pai/pull/742#discussion_r196670423) [](ancestors = 196670423)